lib/puppet/provider/openldap_database/olc.rb

```diff
@@ -131,10 +131,29 @@ def exists?
     @property_hash[:ensure] == :present
   end
 
+  def fetch_index
+arg = ['-b',
+      'cn=config',
+      '-H',
+      "ldap:///???(&(objectClass=olc#{@property_hash[:backend].to_s.capitalize}Config)(olcSuffix=#{@property_hash[:suffix]}))"]
+    $stderr.puts(arg.inspect)
+    slapcat(
+      '-b',
+      'cn=config',
+      '-H',
+      "ldap:///???(&(objectClass=olc#{@property_hash[:backend].to_s.capitalize}Config)(olcSuffix=#{@property_hash[:suffix]}))").split("\n").collect do |line|
+      if line =~ /^olcDatabase: /
+        @property_hash[:index] = line.match(/^olcDatabase: \{(\d+)\}#{@property_hash[:backend]}$/).captures[0].to_i
+      end
+    end
+  end
```

Comment:
Eventually, the whole provider should look different and have way less code. I tried to keep this PR self-contained because the method is quite tricky.